scripts/generate_build_number.sh

```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
+
+# Tweak our build number and export it as a variable for the nuget packing script
+BUILD_VERSION_SUFFIX=""
+BUILD_VERSION_STRING="0.0.0.1"
+
+if [ -n "$BUILD_NUMBER_SUFFIX_OVERRIDE" ]; then 
+    # Allow someone calling us to provide an override to always add a suffix to a given build number by providing an env variable
+    ADJUSTED_BUILD_NUMBER_SUFFIX_OVERRIDE="${BUILD_NUMBER_SUFFIX_OVERRIDE//[^a-z0-9A-Z]/}"
+    BUILD_VERSION_SUFFIX="-$ADJUSTED_BUILD_NUMBER_SUFFIX_OVERRIDE"
+elif [ -n "$BUILD_SOURCEBRANCHNAME" ] && [ "$BUILD_SOURCEBRANCHNAME" != "master" ]; then 
+    # Otherwise append the branch name unless we're the main branch for this repo
+    ADJUSTED_SOURCEBRANCHNAME="${BUILD_SOURCEBRANCHNAME//[^a-z0-9A-Z]/}"
+    BUILD_VERSION_SUFFIX="-$ADJUSTED_SOURCEBRANCHNAME"
+fi
+
+if [ -n "$BUILD_BUILDNUMBER" ]; then
+    BUILD_VERSION_STRING="$BUILD_BUILDNUMBER"
+fi
+
+echo "Original Build Number: $BUILD_BUILDNUMBER"
+echo "Adjusted Build Number: $BUILD_VERSION_STRING$BUILD_VERSION_SUFFIX"
+
+echo "##vso[task.setvariable variable=sanitizedBuildNumber]$BUILD_VERSION_STRING$BUILD_VERSION_SUFFIX"
```

Comment:
This file is copied from FluentUI Apple https://github.com/microsoft/fluentui-apple/blob/master/scripts/generate_build_number.sh

Reply:
Not sure what you mean? This build number will be unique though, yes

Reply:
Basically I was wondering if all platforms are aligned on this build number strategy or we're planning this to be Apple specific. Basically is this file run in publish for all platforms or just Apple?

Reply:
Not planning this to be apple specific. I do wonder however if we want to prefix with the @fluentui/react-native semantic package version. Opened #480 to track